### ISSUE TITLE
Disposing the Hub with AutoSessionTracking gives an error

### ISSUE THREAD

**Issue by lucas-zimerman** [getsentry/sentry-dotnet#1374]
As observed on Unity, when you dispose of a hub and usingAutoSessionTracking, an error will be thrown once the SDK disposes of the SDK hub. You can reproduce the error with the following snippet:

```csharp
void StartSentry()
{
    SentrySdk.Init(o =>
    {
        o.Dsn = "https://eb18e953812b41c3aeb042e666fd3b5c@o447951.ingest.sentry.io/5428537";
        o.AutoSessionTracking = true;
        o.CacheDirectoryPath = "C:/Users/.../AppData/LocalLow/Sentry/Sentry Defenses/Sentry";
        o.Debug = true;
    });
}
StartSentry();
StartSentry();
Console.ReadKey();
```

The following error was received:
```
Error: Failed to send cached envelope: C:/Users/lucas/AppData/LocalLow/Sentry/Sentry Defenses/Sentry\Sentry\132A73C90BBA506000857BC4BFC23E574F1551B5\__processing\1638992610_1647__6590935.envelope, retrying after a delay.
System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at Sentry.Internal.Extensions.StreamExtensions.ReadAllBytesAsync(Stream stream, CancellationToken cancellationToken)+MoveNext() in D:\Development\Online Projects\Sentry\sentry-dotnet\src\Sentry\Internal\Extensions\StreamExtensions.cs:line 18
   at Sentry.Internal.Extensions.StreamExtensions.ReadAllBytesAsync(Stream stream, CancellationToken cancellationToken)+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
   at Sentry.Protocol.Envelopes.Envelope.DeserializeHeaderAsync(Stream stream, CancellationToken cancellationToken) in D:\Development\Online Projects\Sentry\sentry-dotnet\src\Sentry\Envelopes\Envelope.cs:line 188
   at Sentry.Protocol.Envelopes.Envelope.DeserializeHeaderAsync(Stream stream, CancellationToken cancellationToken) in D:\Development\Online Projects\Sentry\sentry-dotnet\src\Sentry\Envelopes\Envelope.cs:line 188
   at Sentry.Protocol.Envelopes.Envelope.DeserializeAsync(Stream stream, CancellationToken cancellationToken) in D:\Development\Online Projects\Sentry\sentry-dotnet\src\Sentry\Envelopes\Envelope.cs:line 212
   at Sentry.Internal.Http.CachingTransport.InnerProcessCacheAsync(String envelopeFilePath, CancellationToken cancellationToken) in D:\Development\Online Projects\Sentry\sentry-dotnet\src\Sentry\Internal\Http\CachingTransport.cs:line 195
   at Sentry.Internal.Http.CachingTransport.InnerProcessCacheAsync(String envelopeFilePath, CancellationToken cancellationToken) in D:\Development\Online Projects\Sentry\sentry-dotnet\src\Sentry\Internal\Http\CachingTransport.cs:line 207
   at Sentry.Internal.Http.CachingTransport.ProcessCacheAsync(CancellationToken cancellationToken) in D:\Development\Online Projects\Sentry\sentry-dotnet\src\Sentry\Internal\Http\CachingTransport.cs:line 159  Debug: Background worker of CachingTransport has shutdown.
```

**Comment by bruno-garcia** [comment]
The SDK is getting Disposed when Init is called again. So the behavior is correct there.
The issue is: Why is it getting Inititalized twice? Init is supposed to be called once in the app lifecycle

**Comment by SimonCropp** [comment]
>  Init is supposed to be called once in the app lifecycle

so should we throw on the second init call?

**Comment by SimonCropp** [comment]
@bruno-garcia @lucas-zimerman  thoughts on the above?

**Comment by vaind** [comment]
> so should we throw on the second init call?

That would be breaking though, wouldn't it?

**Comment by bruno-garcia** [comment]
The SDK should be OK with being disposed without breaking internally though. Once there's "more information" a following Init call is done. The static `SentrySdk` class does an atomic swap between hubs (old to new) so we can flush it and let it go out of scope so GC does it's thing. Since we don't want to Dispose in favor to Flushing only (see #599 ), it's possible finalizer thread will need to clear some unmanaged resources if we happened to allocate one in the previous calls

**Comment by mattjohnsonpint** [comment]
Closing in favor of #2033
